parser.js

```diff
@@ -562,25 +563,27 @@
       }
 
       var atom = parseAtomAndExtendedAtom();
+      var quantifier;
       if (!atom) {
         // Check if a quantifier is following. A quantifier without an atom
         // is an error.
         var pos_backup = pos
-        var quantifier = parseQuantifier() || false;
+        quantifier = parseQuantifier() || false;
```

Comment:
I dislike this because the old code was valid, but I guess most people prefer to hoist `var` as if it was `let` as you did :grimacing:

Reply:
yeh I guess it worked but the `var quantifier` below was shadowing this one

Reply:
I'd prefer `let`/`const` and then there would be no issue here, but not sure if you want to introduce those?

Reply:
Since this package doesn't specify the minimum Node.js version in `package.json`, I'd prefer to stick to `var` so that we don't accidentally break support for some old versions.

Reply:
sounds good to me. I could undo this change and disable the eslint rule for shadowing, but shadowing can lead to mistakes so IMO this is better (if we stick with `var`)